src/nvim/normal.c

```diff
@@ -350,20 +350,23 @@ static int nv_compare(const void *s1, const void *s2)
  */
 void init_normal_cmds(void)
 {
-  int i;
+  assert(NV_CMDS_SIZE <= SHRT_MAX);
 
   /* Fill the index table with a one to one relation. */
-  for (i = 0; i < (int)NV_CMDS_SIZE; ++i)
+  for (short int i = 0; i < (short int)NV_CMDS_SIZE; ++i) {
     nv_cmd_idx[i] = i;
+  }
 
   /* Sort the commands by the command character.  */
-  qsort((void *)&nv_cmd_idx, (size_t)NV_CMDS_SIZE, sizeof(short), nv_compare);
+  qsort(&nv_cmd_idx, NV_CMDS_SIZE, sizeof(short), nv_compare);
 
   /* Find the first entry that can't be indexed by the command character. */
-  for (i = 0; i < (int)NV_CMDS_SIZE; ++i)
-    if (i != nv_cmds[nv_cmd_idx[i]].cmd_char)
+  for (short int i = 0; i < (short int)NV_CMDS_SIZE; ++i) {
+    if (i != nv_cmds[nv_cmd_idx[i]].cmd_char) {
+      nv_max_linear = i - 1;
       break;
-  nv_max_linear = i - 1;
```

Comment:
That's true. I'd say such a thing would never happen with current command table, but I'll change it back for safety.